src/app/content/highlights/components/cardUtils.spec.ts

```diff
@@ -2,49 +2,97 @@ import { Highlight } from '@openstax/highlighter';
+    });
+  });
+
+  describe('getHighlightTopOffset', () => {
+    it('returns undefined if passed container is undefined', () => {
+      expect(getHighlightTopOffset(undefined, { id: 'asd' } as Highlight)).toBeUndefined();
+    });
+
+    it('returns top offset for a highlight', () => {
+      expect(getHighlightTopOffset(
         assertDocument().createElement('div'),
         createMockHighlight() as any as Highlight)
       ).toEqual(0);
+    });
   });
-});
 
-describe('generateUpdatePayload', () => {
-  const highlightId = 'my-id';
-  const oldData = {
-    annotation: 'my annotation',
-    color: HighlightColorEnum.Blue,
-  };
+  describe('generateUpdatePayload', () => {
+    const highlightId = 'my-id';
+    const oldData = {
+      annotation: 'my annotation',
+      color: HighlightColorEnum.Blue,
+    };
 
-  it('handles updating both fields', () => {
-    const update = {color: HighlightColorEnum.Green, annotation: 'hello'};
-    const payload = generateUpdatePayload(oldData, {id: highlightId, ...update});
+    it('handles updating both fields', () => {
+      const update = { color: HighlightColorEnum.Green, annotation: 'hello' };
+      const payload = generateUpdatePayload(oldData, { id: highlightId, ...update });
 
-    expect(payload.preUpdateData.highlight).toEqual(oldData);
-    expect(payload.updatePayload.highlight).toEqual(update);
-  });
+      expect(payload.preUpdateData.highlight).toEqual(oldData);
+      expect(payload.updatePayload.highlight).toEqual(update);
+    });
 
-  it('handles updating only color', () => {
-    const update = {color: HighlightColorEnum.Green};
-    const payload = generateUpdatePayload(oldData, {id: highlightId, ...update});
+    it('handles updating only color', () => {
+      const update = { color: HighlightColorEnum.Green };
+      const payload = generateUpdatePayload(oldData, { id: highlightId, ...update });
 
-    expect(payload.preUpdateData.highlight).toEqual(oldData);
-    expect(payload.updatePayload.highlight).toEqual({...update, annotation: oldData.annotation});
-  });
+      expect(payload.preUpdateData.highlight).toEqual(oldData);
+      expect(payload.updatePayload.highlight).toEqual({ ...update, annotation: oldData.annotation });
+    });
+
+    it('handles updating only annotation', () => {
+      const update = { annotation: 'hi' };
+      const payload = generateUpdatePayload(oldData, { id: highlightId, ...update });
 
-  it('handles updating only annotation', () => {
-    const update = {annotation: 'hi'};
-    const payload = generateUpdatePayload(oldData, {id: highlightId, ...update});
+      expect(payload.preUpdateData.highlight).toEqual(oldData);
+      expect(payload.updatePayload.highlight).toEqual({ ...update, color: oldData.color });
+    });
+  });
 
-    expect(payload.preUpdateData.highlight).toEqual(oldData);
-    expect(payload.updatePayload.highlight).toEqual({...update, color: oldData.color});
+  describe('updateStackedCardsPositions', () => {
```

Comment:
I think it would make more sense to mock the actual behavior in CardWrapper.spec.tsx by passing for example 3 highlights and putting one of them into the collapsed container

Reply:
there are tests that checks if cards positions were updated so you can use them as references

Reply:
```
jest.mock('./cardUtils', () => ({
  ...jest.requireActual('./cardUtils'),
  getHighlightOffset: jest.fn(() => ({ top: 100, bottom: 100 })),
}));
``` 
There is a `getHighlightOffset` mock in `CardWrapper.spec.tsx`.

Reply:
I could just mock `getHighlightOffset` again but it won't let me test the actual behavior of group of highlights.

Reply:
you could try to indetify which tests needs this mock and mock this function only in those tests